tests/installation/bootloader_hyperv.pm

```diff
@@ -145,12 +145,16 @@ sub run {
 
     my ($winserver, $winver, $vmguid);
 
-    $winver = $svirt->get_cmd_output("cmd /C ver");
-    if (grep { /Microsoft Windows \[Version 6.3.*\]/ } $winver) {
-        $winserver = 2012;
+    $winver = $svirt->get_cmd_output("$ps (Get-WmiObject Win32_OperatingSystem).Version");
+    $winver =~ s/^\s+|\s+$//g;
+    if ($winver eq '6.3.9600') {
+        $winserver = '2012R2';
     }
-    elsif (grep { /Microsoft Windows \[Version 10.0.*\]/ } $winver) {
-        $winserver = 2016;
+    elsif ($winver eq '10.0.14393') {
```

Comment:
so does this check

Reply:
For example, both build number 14393 and 16299 are Windows Server 2016 (Reference [3]), checking 10.0.14* only is not enough.
Windows Server 2016, Version 1607	10.0	14393
Windows Server 2016, Version 1709	10.0	16299